src/backend/api/reservations.js

```diff
@@ -1,56 +1,65 @@
+const { Router } = require("express");
 const express = require("express");
-const app = express();
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/", async (req, res) => {
     try {
-        const reservations = await knex("reservation").select("id");
-        response.json(reservations);
+        const reservations = await knex("reservations");
+        res.json(reservations);
     } catch (error) {
         throw error;
     }
 });
 
 router.post("/", async (req, res) => {
+    console.log(req.body)
     try {
-        const newReservation = await knex("reservation").insert(req.body);
+        const newReservation = await knex("reservations").insert(request.body);
         res.json(newReservation);
+
     } catch (error) {
         throw error;
     }
 });
 
 router.get("/:id", async (req, res) => {
     try {
-        const reservationById = await knex("reservation").where(req.params);
-        res.json(reservationById);
+        const reservationById = await knex("reservations")
+            .where({
+                id: parseInt(req.params.id)
+            })
+        res.json(reservationById)
     } catch (error) {
         throw error;
     }
 });
 
 router.put("/:id", async (req, res) => {
     try {
-        const updateReservationId = await knex("reservation")
-            .where(req.params)
-            .update(req.body);
-        res.json(updateReservationId);
+        const updateReservationsId = await knex("reservations")
+            .where({
+                id: req.params.id
+            })
+            .update(request.body)
+        res.json(updateReservationsId)
     } catch (error) {
         throw error;
     }
 });
 
 router.delete("/:id", async (req, res) => {
     try {
-        const deleteById = await knex("reservation")
-            .where(req.params)
-            .del();
-        res.json(deleteById);
+        const deleteById = await knex("reservations")
+            .where({
+                id: req.params.id
+            }).del()
+        res.json(deleteById)
```

Comment:
I would change the endpoint urls for delete and edit, instead of just changing the method.
If by mistake you pass the wrong method when you want to fetch you will delete a reservation...
Just a thought 🙂